lib/block-supports/layout.php

```diff
@@ -608,7 +628,17 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 
 	$class_names        = array();
 	$layout_definitions = gutenberg_get_layout_definitions();
-	$container_class    = wp_unique_id( 'wp-container-' );
+
+	/*
+	* We use an incremental ID that is independent per prefix to make sure that
+	* rendering different numbers of blocks doesn't affect the IDs of other
+	* blocks. We need this to make the CSS class names stable across paginations
+	* for features like the enhanced pagination of the Query block.
+	*/
+	$container_class = gutenberg_incremental_id_per_prefix(
+		'wp-container-' .
+		str_replace( '/', '--', $block['blockName'] ) . '-'
```

Comment:
According to [the block name requirements](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#name):

> Note: A block name can only contain lowercase alphanumeric characters, dashes, and at most one forward slash to designate the plugin-unique namespace prefix. It must begin with a letter.

Reply:
Rather than a manual `str_replace` would it be worth using `sanitize_title` for consistency, or is it quicker to use the `str_replace`?

Reply:
I'm not sure what's quicker. Sanitize title would replace the slash with a single dash instead of double, but that's not necessarily a bad outcome because we're not using double dashes in these classnames anyway.

Reply:
I didn't know about `sanitize_title`. I'll make the change 👍